packages/opentelemetry-plugin-postgres/opentelemetry-plugin-pg-pool/src/pg-pool.ts

```diff
@@ -13,3 +13,102 @@
  * limitations under the License.
  */
+
+import { BasePlugin } from '@opentelemetry/core';
+import { SpanKind } from '@opentelemetry/types';
+import { AttributeNames } from './enums';
+import * as shimmer from 'shimmer';
+import * as pgPoolTypes from 'pg-pool';
+import * as pgTypes from 'pg';
+import { PostgresPoolPluginOptions, PgPoolCallback } from './types';
+
+export class PostgresPoolPlugin extends BasePlugin<typeof pgPoolTypes> {
+  protected _config: PostgresPoolPluginOptions;
+
+  static readonly COMPONENT = 'pg-pool';
+
+  readonly supportedVersions = ['^2.0.7'];
+
+  constructor(readonly moduleName: string) {
+    super();
+    this._config = {};
+  }
+
+  protected patch(): typeof pgPoolTypes {
+    if (this._moduleExports.prototype.connect) {
+      shimmer.wrap(
+        this._moduleExports.prototype,
+        'connect',
+        this._getPoolConnectPatch() as never
+      );
+    }
+    return this._moduleExports;
+  }
+
+  protected unpatch(): void {
+    if (this._moduleExports.prototype.connect) {
+      shimmer.unwrap(this._moduleExports.prototype, 'connect');
+    }
+  }
+
+  private _getPoolConnectPatch() {
+    const plugin = this;
+    return (originalConnect: typeof pgPoolTypes.prototype.connect) => {
+      plugin._logger.debug(
+        `Patching ${PostgresPoolPlugin.COMPONENT}.prototype.connect`
+      );
+      return function connect(
+        this: pgPoolTypes<pgTypes.Client>,
+        callback?: PgPoolCallback
+      ) {
+        // setup span
+        const span = plugin._tracer.startSpan(
+          `${PostgresPoolPlugin.COMPONENT}.connect`,
+          {
+            kind: SpanKind.CLIENT,
+            parent: plugin._tracer.getCurrentSpan() || undefined,
+            attributes: {
+              [AttributeNames.COMPONENT]: PostgresPoolPlugin.COMPONENT,
+            },
+          }
+        );
+
+        if (callback) {
+          callback = plugin._tracer.bind(callback);
+        }
+
+        const connectResult: unknown = originalConnect.call(
+          this,
+          callback as never
+        );
+
+        // No callback was provided, return a promise instead
+        if (connectResult instanceof Promise) {
+          const connectResultPromise = connectResult as Promise<unknown>;
+          return plugin._tracer.bind(
+            connectResultPromise
+              .then((result: any) => {
+                // Resturn a pass-along promise which ends the span and then goes to user's orig resolvers
+                return new Promise((resolve, _) => {
+                  span.end();
+                  resolve(result);
+                });
+              })
+              .catch((error: Error) => {
+                return new Promise((_, reject) => {
+                  span.end();
+                  reject(error);
+                });
+              })
+          );
+        }
+
+        // Else a callback was provided, so just return the result
+        span.end();
```

Comment:
For this case, I think this span will be ended immediately -- before the network calls finish. For callbacks scenarios, I think we should patch the callback and end the span (and set the corresponding status) once the callback is called. Similar to pg plugin [`utils.patchCallback`](https://github.com/open-telemetry/opentelemetry-js/blob/f47c2def6510a4845d84805a7f611a42fabd1fb0/packages/opentelemetry-plugin-postgres/opentelemetry-plugin-pg/src/utils.ts#L161-L181)